src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -126,118 +126,6 @@ namespace {
     return true;
   }
 
-  void check_spatial_eq(SpatialPooler sp1, SpatialPooler sp2)
-  {
-    UInt numColumns = sp1.getNumColumns();
-    UInt numInputs = sp2.getNumInputs();
-
-    ASSERT_TRUE(sp1.getNumColumns() == sp2.getNumColumns());
-    ASSERT_TRUE(sp1.getNumInputs() == sp2.getNumInputs());
-    ASSERT_TRUE(sp1.getPotentialRadius() ==
-              sp2.getPotentialRadius());
-    ASSERT_TRUE(sp1.getPotentialPct() == sp2.getPotentialPct());
-    ASSERT_TRUE(sp1.getGlobalInhibition() ==
-              sp2.getGlobalInhibition());
-    ASSERT_TRUE(sp1.getNumActiveColumnsPerInhArea() ==
-              sp2.getNumActiveColumnsPerInhArea());
-    ASSERT_TRUE(almost_eq(sp1.getLocalAreaDensity(),
-              sp2.getLocalAreaDensity()));
-    ASSERT_TRUE(sp1.getStimulusThreshold() ==
-              sp2.getStimulusThreshold());
-    ASSERT_TRUE(sp1.getDutyCyclePeriod() == sp2.getDutyCyclePeriod());
-    ASSERT_TRUE(almost_eq(sp1.getBoostStrength(), sp2.getBoostStrength()));
-    ASSERT_TRUE(sp1.getIterationNum() == sp2.getIterationNum());
-    ASSERT_TRUE(sp1.getIterationLearnNum() ==
-              sp2.getIterationLearnNum());
-    ASSERT_TRUE(sp1.getSpVerbosity() == sp2.getSpVerbosity());
-    ASSERT_TRUE(sp1.getWrapAround() == sp2.getWrapAround());
-    ASSERT_TRUE(sp1.getUpdatePeriod() == sp2.getUpdatePeriod());
-    ASSERT_TRUE(almost_eq(sp1.getSynPermTrimThreshold(),
-              sp2.getSynPermTrimThreshold()));
-    cout << "check: " << sp1.getSynPermActiveInc() << " " <<
-      sp2.getSynPermActiveInc() << endl;
-    ASSERT_TRUE(almost_eq(sp1.getSynPermActiveInc(),
```

Comment:
TODO: many uses of `almost_eq()` can be replaced with gtest native `EXCEPT_NEAR(a, b, diff);` and the function can be **removed** from SPTest class.
https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md

> Although, for the comparisons we need a boolean, not an assert; so we'd have to keep the function?

We'll just implement the small almost_eq method ; see "twice" in the example
#1384 (comment)